build/source/dshare/outpt_stat.f90

```diff
@@ -132,71 +137,75 @@ subroutine calc_stats(meta,stat,tdata,iStep,err,message)
-   if (meta%varType.ne.iLookVarType%outstat) cycle ! only calculate stats for scalars
-   select case(iStat)                              ! act depending on the statistic
-    case (iLookStat%totl)                          ! summation over period
-     tstat(iStat) = 0                              ! resets stat at beginning of period
-    case (iLookStat%mean)                          ! mean over period
-     tstat(iStat) = 0.
-    case (iLookStat%vari)                          ! variance over period
-     tstat(iStat) = 0                              ! resets E[X^2] term in var calc
-     tstat(maxVarStat+1) = 0                       ! resets E[X]^2 term
-    case (iLookStat%mini)                          ! minimum over period
-     tstat(iStat) = huge(tstat(iStat))             ! resets stat at beginning of period
-    case (iLookStat%maxi)                          ! maximum over period
-     tstat(iStat) = -huge(tstat(iStat))            ! resets stat at beginning of period
-    case (iLookStat%mode)                          ! mode over period (does not work)
-     tstat(iStat) = -9999.
+ do iFreq=1,maxVarFreq                              ! loop through output statistics
+  if(resetStats(iFreq))then                         ! flag to reset statistics
+   if(meta%statIndex(iFreq)==integerMissing) cycle  ! don't bother if output frequency is not desired for a given variab;e
+   if(meta%varType/=iLookVarType%outstat) cycle     ! only calculate stats for scalars
+   select case(meta%statIndex(iFreq))               ! act depending on the statistic
+    ! -------------------------------------------------------------------------------------
+    case (iLookStat%totl)                           ! * summation over period                  
+     tstat(iFreq) = 0._dp                           !     - resets stat at beginning of period
+    case (iLookStat%mean)                           ! * mean over period                       
+     tstat(iFreq) = 0._dp                           !     - resets stat at beginning of period
+    case (iLookStat%vari)                           ! * variance over period                   
+     tstat(iFreq) = 0._dp                           !     - resets E[X^2] term in var calc    
+     tstat(maxVarFreq+iFreq) = 0._dp                !     - resets E[X]^2 term                 
+    case (iLookStat%mini)                           ! * minimum over period                    
+     tstat(iFreq) = huge(tstat(iFreq))              !     - resets stat at beginning of period 
+    case (iLookStat%maxi)                           ! * maximum over period                    
+     tstat(iFreq) = -huge(tstat(iFreq))             !     - resets stat at beginning of period 
+    case (iLookStat%mode)                           ! * mode over period (does not work)       
+     tstat(iFreq) = realMissing
+    ! -------------------------------------------------------------------------------------
```

Comment:
Can you add a default to trap any errors (it should never get there, but coding errors happen and this would be easier to catch them)?

Reply:
Done